scripts/tools/metricProgrammable.yml

```diff
@@ -0,0 +1,490 @@
+#
+# See YaML.md for syntax definition
+#
+--- #--------------------------------------------------------------------------
+type: header
+desc: "Intel $OneApi Level-Zero Tool Experimental Extension for Programmable Metrics"
+version: "1.8"
+--- #--------------------------------------------------------------------------
+type: macro
+desc: "Programmable Metrics Experimental Extension Name"
+version: "1.8"
+name: $T_PROGRAMMABLE_METRICS_EXP_NAME
+value: '"$XT_experimental_programmable_metrics"'
+--- #--------------------------------------------------------------------------
+type: enum
+desc: "Programmable Metrics Experimental Extension Version(s)"
+version: "1.8"
+name: $t_metric_programmable_exp_version_t
+etors:
+    - name: "1_0"
+      value: "$X_MAKE_VERSION( 1, 0 )"
+      desc: "version 1.0"
+--- #--------------------------------------------------------------------------
+type: macro
+desc: "Maximum count of characters in export data element name"
+version: "1.8"
+name: $T_MAX_PROGRAMMABLE_METRICS_ELEMENT_NAME_EXP
+value: "256"
+--- #--------------------------------------------------------------------------
+type: macro
+desc: "Maximum export data element description string size"
+version: "1.8"
+name: $T_MAX_PROGRAMMABLE_METRICS_ELEMENT_DESCRIPTION_EXP
+value: "256"
+--- #--------------------------------------------------------------------------
+type: macro
+desc: "Maximum metric programmable name string size"
+version: "1.8"
+name: $T_MAX_METRIC_PROGRAMMABLE_NAME_EXP
+value: "128"
+--- #--------------------------------------------------------------------------
+type: macro
+desc: "Maximum metric programmable description string size"
+version: "1.8"
+name: $T_MAX_METRIC_PROGRAMMABLE_DESCRIPTION_EXP
+value: "128"
+--- #--------------------------------------------------------------------------
+type: macro
+desc: "Maximum metric programmable component string size"
+version: "1.8"
+name: $T_MAX_METRIC_PROGRAMMABLE_COMPONENT_EXP
+value: "128"
+--- #--------------------------------------------------------------------------
+type: macro
+desc: "Maximum metric programmable parameter string size"
+version: "1.8"
+name: $T_MAX_METRIC_PROGRAMMABLE_PARAMETER_NAME_EXP
+value: "128"
+--- #--------------------------------------------------------------------------
+type: macro
+desc: "Maximum value information string size"
+version: "1.8"
+name: $T_MAX_VALUE_INFO_CSTRING_EXP
+value: "128"
+--- #--------------------------------------------------------------------------
+type: handle
+desc: "Handle of metric programmable's object"
+version: "1.8"
+class: $tMetricProgrammable
+name: "$t_metric_programmable_exp_handle_t"
+--- #--------------------------------------------------------------------------
+type: struct
+desc: "Metric Programmable properties queried using $tMetricProgrammableGetPropertiesExp"
+version: "1.8"
+class: $tMetricProgrammable
+name: $t_metric_programmable_exp_properties_t
+base: $t_base_properties_t
+members:
+    - type: char
```

Comment:
all interfaces in L0 have an owner (a driverHandle, a deviceHandle, a metric handle, etc), what is the one for this interface?

Reply:
Could you please clarify more.

t_metric_programmable_exp_properties_t is a structure and not an interface/API